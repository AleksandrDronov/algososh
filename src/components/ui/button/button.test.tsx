import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";
import { Button } from "./button";

describe("Кнопка рендерится без ошибок", () => {
  it("Кнопка c текстом", () => {
    const button = renderer.create(<Button text="Кнопка" />);

    expect(button).toMatchSnapshot();
  });
  it("Кнопка без текста", () => {
    const button = renderer.create(<Button />);

    expect(button).toMatchSnapshot();
  });
  it("Кнопка заблокирована", () => {
    const button = renderer.create(<Button disabled={true} />);

    expect(button).toMatchSnapshot();
  });
  it("Кнопка с индикацией загрузки", () => {
    const button = renderer.create(<Button isLoader={true} />);

    expect(button).toMatchSnapshot();
  });
  it("Колбек кнопки", () => {
    window.alert = jest.fn();
    const onClick = () => alert("Кнопка нажата");
    render(<Button onClick={onClick} text={"Кнопка"}/>);
    const link = screen.getByText("Кнопка");
    fireEvent.click(link);
    expect(window.alert).toHaveBeenCalledWith("Кнопка нажата");
  });
});
