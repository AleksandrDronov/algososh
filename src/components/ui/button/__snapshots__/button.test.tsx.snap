// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Кнопка рендерится без ошибок Кнопка c текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Кнопка
  </p>
</button>
`;

exports[`Кнопка рендерится без ошибок Кнопка без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Кнопка рендерится без ошибок Кнопка заблокирована 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Кнопка рендерится без ошибок Кнопка с индикацией загрузки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;
